src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -907,6 +907,382 @@ public static async Task ZipArchive_InvalidHuffmanData()
+                {
+                    ZipArchiveEntry newEntry = updatedArchive.CreateEntry("second.bin", CompressionLevel.NoCompression);
+
+                    // Add data to the new entry
+                    using (Stream entryStream = newEntry.Open())
+                    {
+                        entryStream.Write(new byte[] { 0x00, 0x01, 0x02, 0x03 });
+                    }
+                }
+
+                byte[] updatedContents = updatedStream.ToArray();
+                // Verify that the local file header and the central directory headers have both been rewritten, and both have
+                // the correct value.
+                int updatedLocalVersionToExtract = updatedContents[4];
+                int updatedCentralDirectoryVersionToExtract = updatedContents[101];
+
+                Assert.Equal(20, updatedCentralDirectoryVersionToExtract);
+                Assert.Equal(20, updatedLocalVersionToExtract);
+
+                updatedStream.Seek(0, SeekOrigin.Begin);
+                // Following an update of the ZipArchive, reopen it in read-only mode. Make sure that both entries are correct.
+
+                using (ZipArchive updatedArchive = new ZipArchive(updatedStream, ZipArchiveMode.Read, true))
+                {
+                    Assert.Equal(2, updatedArchive.Entries.Count);
+
+                    ZipArchiveEntry firstEntry = updatedArchive.Entries[0];
+                    ZipArchiveEntry secondEntry = updatedArchive.Entries[1];
+
+                    Assert.Equal("first.bin", firstEntry.Name);
+                    Assert.Equal(10, firstEntry.Length);
+
+                    Assert.Equal("second.bin", secondEntry.Name);
+                    Assert.Equal(4, secondEntry.Length);
+
+                    using (Stream entryStream = firstEntry.Open())
+                    {
+                        byte[] uncompressedBytes = new byte[firstEntry.Length];
+                        int bytesRead = entryStream.Read(uncompressedBytes);
+
+                        Assert.Equal(10, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09 }, uncompressedBytes);
+                    }
+
+                    using (Stream entryStream = secondEntry.Open())
+                    {
+                        byte[] uncompressedBytes = new byte[secondEntry.Length];
+                        int bytesRead = entryStream.Read(uncompressedBytes);
+
+                        Assert.Equal(4, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03 }, uncompressedBytes);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(0x00)]
+        [InlineData(0x14)]
+        public void ZipArchive_InvalidExtraFieldData(byte validVersionToExtract)
+        {
+            byte[] invalidExtraFieldData = s_invalidExtraFieldData.AsSpan().ToArray();
```

Comment:
This should be enough:
```suggestion
            byte[] invalidExtraFieldData = s_invalidExtraFieldData;
```